packages/@tinacms/fields/src/NumberInput.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import { Input } from './Input'
+
+interface NumberFieldProps {
+  label: string
+  name: string
+  component: string
+  steps: number | string
+}
+
+export interface NumberProps {
+  name: string
+  input: any
+  field: NumberFieldProps
+  disabled?: boolean
+  steps: number | string
+}
+
+export const NumberInput: React.FC<NumberProps> = ({ input, field, steps }) => (
```

Comment:
My one complaint right now is that the `NumberInput` takes in the full field props. 

The `@tinacms/fields` are meant to be just plain inputs that could be used anywhere. They shouldn't know about `field` or anything. They should just accept the input props i.e. `<NumberField {...input} />`

Checkout the `TextField` and `TextFieldPlugin` to see what I mean. The [TextField](https://github.com/tinacms/tinacms/blob/master/packages/@tinacms/fields/src/TextField.ts#L32-L34) is just a styled input.

The [TextFieldPlugin](https://github.com/tinacms/tinacms/blob/master/packages/tinacms/src/plugins/fields/TextFieldPlugin.tsx#L24-L29) is where the values are pulled off of `field` and passed to the `TextField`.

Reply:
Alright ... You don't want the user to be able to set the steps on the field? Cause I'm grabbing these from the field prop.

Reply:
Yeah exactly, would be passed the same way `placeholder` is passed from the `TextFieldPlugin` to the `TextField`

Reply:
Something like:
```
export const NumberField = wrapFieldsWithMeta<
  { step: string | number },
  InputProps
>(({ input, field }) => (
  <BaseNumberField {...input} step={field.step} />
))

Reply:
I'm onto it ... Haven't been in the codebase for a while so have to update my brain to go through what is doing what. =)

Reply:
Hah absolutely. I can do it later as well if you're busy.

Reply:
Hopefully it is like you want it now. =) Just pushed my changes ... Been coding for 12 hours today so can't really think now ...